### PR TITLE
Priority event listeners and event abort

### PR DESCRIPTION
**Pull request by floryst**
This adds priorities to event listeners, as well as capability for event listeners to abort the continuation/propagation of a vtkjs event.

### CODE REVIEW

File: Sources/macro.js [hunk 1]
```diff
@@ -722,30 +733,37 @@ export function event(publicAPI, model, eventName) {
...
       vtkErrorMacro('instance deleted - cannot call any method');
       return null;
     }
 
-    const index = callbacks.length;
-    callbacks.push(callback);
-    return on(index);
+    const callbackID = curCallbackID++;
+    callbacks.push([callbackID, callback, priority]);
+    callbacks.sort(([, cb1, pri1], [, cb2, pri2]) => pri2 - pri1);
```
Comment (jourdain) [inline Sources/macro.js]: your sort does not use cb1 and cb2...

Comment (jourdain) [reply Sources/macro.js]: what about doing it like that

```
callbacks.sort((a, b) => b[2] - a[2]);
```

With maybe the function be only defined once so it does not get created at each call?

Comment (floryst) [reply Sources/macro.js]: The comparator is small enough do not need a separate function. Will there be a measurable performance hit? Maybe if on* gets called a lot...

Comment (jourdain) [reply Sources/macro.js]: I agree, and I'm not expecting perf hit.

Comment (floryst) [reply Sources/macro.js]: :+1:

**Comment by floryst** [comment]
Are there any docs for macro.js that I can update?

**Comment by jourdain** [comment]
No we have the big comment on each macro function. But it would be nice to have a markdown for it.

**Comment by jourdain** [comment]
Maybe a full markdown could make sense next to https://kitware.github.io/vtk-js/docs/develop_class.html which can be mentioned in that class doc as well.

**Comment by floryst** [comment]
Hm, I was thinking an `api.md` sort of file, but the `develop_class.md` could link to it. How does that sound?

**Comment by jourdain** [comment]
api.md would be good, but macro is special in the sense that's not a class. Hence I was thinking `./Documentation/content/docs/develop_macro.md` instead that we can put in the menu of `Docs` as well as referencing it inside the `develop_class.md`.

**Comment by jourdain** [comment]
Should we merge it? Or do you want to push the doc at the same time?

**Pull request merged**